lib/pxlogger.js

```diff
@@ -4,27 +4,35 @@
  */
 
 class PxLogger {
-    constructor() {
-        this.debugMode = false;
-        this.appId = '';
-    }
+  constructor() {
+    this.debugMode = false;
+    this.appId = "";
+    this.logger = console;
+  }
 
-    init(pxConfig) {
-        this.debugMode = pxConfig.conf.DEBUG_MODE;
-        this.appId = pxConfig.conf.PX_APP_ID;
-    }
+  setCustomLogger(customLogger) {
```

Comment:
I can't reply to the other comment string for some reason, but I read through it. It's unfortunate that there is a circular dependency between `PxLogger` and `PxConfig`. 

That said, given that, I can think of a few potential paths forward here:
* I'm not sure there is a ton of value working in the `CUSTOM_LOGGER` value into the `PxConfig` class, as it's only used in the logger `init` function, and at the point of running that, we've already initialized things via the `setCustomLogger` function. At that point, you could actually kill the `CUSTOM_LOGGER` additions in `PxConfig` all together and call `setCustomLogger` right after instantiation [here](https://github.com/PerimeterX/perimeterx-node-core/blob/master/lib/pxenforcer.js#L18).
* Building off of that, we could just kill the `setCustomLogger` function all together and pass the custom logger into the constructor and set the logger immediately:
  ```js
  constructor(params) {
    this.debugMode = false;
    this.appId = "";
    this.logger = console || params.logger;
  }
  ```

**I'd prefer the constructor option of the two above. It streamlines this PR completely.**

But it may make sense to take this one more step further. To your point, there are `debug` log calls in the PxConfig initialization [here](https://github.com/PerimeterX/perimeterx-node-core/blob/master/lib/pxconfig.js#L65), but the default `debugMode` in the logger is set to `false` in the constructor, so those will never print out. Ever. Instead, I would think we should actually initialize all constructor values off of params vs. PxConfig and push default values into the logger constructor to make it self-sufficient and eliminate the circular dependencies + `init` method all together. We will need PX team to weigh in on this, but it feels like the best direction overall given the state of this codebase and actually resolves another bug.

Reply:
went with the constructor route 👍